src/kOS.Safe/Screen/ScreenBufferLine.cs

```diff
@@ -67,13 +68,22 @@ public ScreenBufferLine(int size)
         /// </summary>
         public void ArrayCopyFrom(char[] source, int sourceStart, int destinationStart, int length = -1)
         {
-            Array.Copy(source, sourceStart, charArray, destinationStart, (length >= 0 ? length : source.Length));
+            int maxSafeLength = (charArray.Length - destinationStart);
+
+            int effectiveLength = (length < 0 ? source.Length : length);
+            if (effectiveLength > maxSafeLength)
+            {
+                effectiveLength = maxSafeLength;
+            }
+
+            Array.Copy(source, sourceStart, charArray, destinationStart, effectiveLength);
             TouchTime();
         }
 
         /// <summary>
         /// Perform the Array.Copy() function.
-        /// Copies a subrange of values from a source array into this array.
+        /// Copies a subrange of values from a source array into this array,
+        /// with safety checks to truncate the copy length if it's out of range.
```

Comment:
It is correct because if you look inside, it makes a call to the other method where I *diid* change it.  I was just documenting it properly for the xml docs so the person calling this method knows it will happen, without caring about the fact that it's actually happening due to what changed in a method call nested inside.  From the "public" viewpoint, the fact that that's why the behavior changed isn't relevant.